ar.c

```diff
@@ -143,6 +150,50 @@ ar_touch (const char *name)
     struct file *arfile;
     arfile = enter_file (strcache_add (arname));
     f_mtime (arfile, 0);
+# ifdef _GUARDIAN_TARGET
+    if (!arfile->updated) {
+      int fd;
+      struct file *file = arfile;
+
+      EINTRLOOP (fd, open (arfile->name, O_RDWR | O_CREAT, 0666));
+      if (fd < 0)
+        TOUCH_ERROR ("touch: open: ");
+      else
+        {
+          struct stat statbuf;
+          char buf = 'x';
+          int e;
+
+          EINTRLOOP (e, fstat (fd, &statbuf));
+          if (e < 0)
+            TOUCH_ERROR ("touch: fstat: ");
+          /* Rewrite character 0 same as it already is.  */
+          EINTRLOOP (e, read (fd, &buf, 1));
+          if (e < 0)
+            TOUCH_ERROR ("touch: read: ");
+          {
+            off_t o;
+            EINTRLOOP (o, lseek (fd, 0L, 0));
+            if (o < 0L)
+              TOUCH_ERROR ("touch: lseek: ");
+          }
+          EINTRLOOP (e, write (fd, &buf, 1));
+          if (e < 0)
+            TOUCH_ERROR ("touch: write: ");
+
+          /* If file length was 0, we just changed it, so change it back.  */
+          if (statbuf.st_size == 0)
+            {
```

Comment:
here too

Reply:
I know. I am vexed as well. If I could just reformat the whole mess it would make me happier, but that will make reconciling and applying changes in the next release much harder.